### PR TITLE
[DI] Enable env placeholders during extension prepend

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | 4.2
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no     
| Deprecations? | no
| Tests pass?   | yes    
| Fixed tickets | #31608 
| License       | MIT
| Doc PR        | https://github.com/symfony/symfony-docs/pull/11599

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/ContainerBuilder.php [hunk 1]
```diff
@@ -695,7 +695,7 @@ public function getExtensionConfig($name)
             $this->extensionConfigs[$name] = [];
         }
 
-        return $this->extensionConfigs[$name];
+        return $this->parameterBag->resolveValue($this->extensionConfigs[$name]);
```
Comment (ro0NL) [inline src/Symfony/Component/DependencyInjection/ContainerBuilder.php]: the fix evolves around this change, but is causing different results in `testOverriddenEnvsAreMerged`. Im not sure it's expected ... cc @nicolas-grekas 

However it's needed for an out-of-the-box fix

**Comment by ro0NL** [comment]
I figure the best way it too create another tmp container, being passed to `prepend()` so we can separately check for overridden envs. Much like what's happening during `load()`. But it's quite a step in terms of changing behavior ... :thinking:

File: src/Symfony/Component/DependencyInjection/ContainerBuilder.php (hunk 1 above)
Comment (ro0NL) [reply src/Symfony/Component/DependencyInjection/ContainerBuilder.php]: see https://github.com/symfony/symfony-docs/pull/11599

**Comment by Simperfit** [comment]
@ro0NL this can be ready for review ? it seems that the fix works, can we review it ?

**Comment by ro0NL** [comment]
@Simperfit conceptually it works, but im not sure about the behavior change in `testOverriddenEnvsAreMerged`

**Comment by javiereguiluz** [comment]
@ro0NL what should we do about this? Do you want to make it mergeable? Have you changed your mind and prefer to change it or not to merge? Thanks.

**Comment by ro0NL** [comment]
@javiereguiluz im not sure yet :) i dont have the time to continue this topic for now. The issue is complex, but has no real priority IMHO.

**Comment by fabpot** [comment]
Closing as this PR is draft and stale :)

**Pull request closed without merging**